File: exe8.py
class Empregado:
    def __init__(self, nome, cargo, salario):
        self.nome = nome
        self.cargo = cargo
        self.salario = salario

    def __str__(self):
        return f"Nome: {self.nome}, Cargo: {self.cargo}, Salário: R$ {self.salario:.2f}"

class Empresa:
    def __init__(self, nome):
        self.nome = nome
        self.empregados = []

    def adicionar_empregado(self, empregado):
        self.empregados.append(empregado)

    def listar_empregados(self):
        print(f"Empregados da empresa {self.nome}:")
        for empregado in self.empregados:
            print(empregado)

if __name__ == "__main__":
    empresa = Empresa("JVA Solutions")

    emp1 = Empregado("João Silva", "Desenvolvedor", 5000)
    emp2 = Empregado("Maria Oliveira", "Gerente", 8000)

    empresa.adicionar_empregado(emp1)
    empresa.adicionar_empregado(emp2)

    empresa.listar_empregados()

File: exe8.go
package main

import (
    "fmt"
)

type Empregado struct {
    Nome   string
    Cargo  string
    Salario float64
}

type Empresa struct {
    Nome      string
    Empregados []Empregado
}

func (e *Empresa) AdicionarEmpregado(empregado Empregado) {
    e.Empregados = append(e.Empregados, empregado)
}

func (e Empresa) ListarEmpregados() {
    fmt.Printf("Empregados da empresa %s:\n", e.Nome)
    for _, empregado := range e.Empregados {
        fmt.Printf("Nome: %s, Cargo: %s, Salário: R$ %.2f\n", empregado.Nome, empregado.Cargo, empregado.Salario)
    }
}

func main() {
    empresa := Empresa{Nome: "JVA Solutions"}

    emp1 := Empregado{Nome: "João Silva", Cargo: "Desenvolvedor", Salario: 5000}
    emp2 := Empregado{Nome: "Maria Oliveira", Cargo: "Gerente", Salario: 8000}

    empresa.AdicionarEmpregado(emp1)
    empresa.AdicionarEmpregado(emp2)

    empresa.ListarEmpregados()
}
